.github/workflows/dbp_pr_checks.yml

```
@@ -0,0 +1,232 @@
+name: DBP - PR Checks
+
+defaults:
+  run:
+    working-directory: SharedPackages/DataBrokerProtectionShared
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'SharedPackages/DataBrokerProtectionShared/**'
+  pull_request:
+    paths:
+      - '.github/**'
+      - 'SharedPackages/DataBrokerProtectionShared/**'
+
+jobs:
+
+  unit-tests:
+
+      name: Run unit tests (macOS)
+
+      runs-on: macos-15
+      timeout-minutes: 30
+
+      outputs:
+        commit_author: ${{ steps.fetch_commit_author.outputs.commit_author }}
+
+      steps:
+
+      - name: Check out the code
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Set cache key hash
+        run: |
+          has_only_tags=$(jq '[ .pins[].state | has("version") ] | all' Package.resolved)
+          if [[ "$has_only_tags" == "true" ]]; then
+            # hashFiles does not respect the working directory, so the full path must be specified
+            echo "cache_key_hash=${{ hashFiles('SharedPackages/DataBrokerProtectionShared/Package.resolved') }}" >> $GITHUB_ENV
+          else
+            echo "Package.resolved contains dependencies specified by branch or commit, skipping cache."
+          fi
+
+      - name: Cache SPM
+        if: env.cache_key_hash
+        uses: actions/cache@v4
+        with:
+          path: |
+            SharedPackages/DataBrokerProtectionShared/.build/artifacts
+            SharedPackages/DataBrokerProtectionShared/.build/checkouts
+            SharedPackages/DataBrokerProtectionShared/.build/repositories
+            SharedPackages/DataBrokerProtectionShared/.build/workspace-state.json
+          key: ${{ runner.os }}-bsk-spm-${{ env.cache_key_hash }}
```

We can use a different cache key here just for clarity - as it's set up now this shouldn't cause any issues since the Package.resolved version is obviously different to BSK, but something like this would be clearer:

```suggestion
          key: ${{ runner.os }}-dbp-spm-${{ env.cache_key_hash }}
```